### PR TITLE
refactor chat mentions widget

### PR DESCRIPTION
**Pull request by gtalha07**
This attempts to remap the existing ```flutter_mentions``` widget to ```auto_complete_trigger``` along with reactive updates to avatars by cleaning up mention providers and instead directly feed ```avatarInfo``` to keep up-to-date with underlying avatar data, essentially optimising the list result even if its big .i.e. we're talking over +100 members.

The reason for switching this plugin to a more simplified version ```auto_complete_trigger``` is that it directly nests the ```TextEditingController```, giving us more autonomy and thereby and more control over tweaking stuff behind where ```flutter_tagger``` lacks. 

### What are the next steps??
To keep PR short and concise for quick review, this doesn't bring any changes or improvements to current mention UI. However these course of features are set to be iterated on our version of ```auto_complete_trigger```.

- Bringing styles on trigger text .i.e. overriding ```TextEditingController``` to make `TextSpans` on demand.
- Up-down arrows functionality to select items and hit enter to add it (Desktop).
- Hashtag trigger for room mentions / Emoji text triggers.

Here's a preview which shows how the list reactively updates it underlying ```AvatarInfo``` data upon mention search.

https://github.com/user-attachments/assets/925628f0-21ab-4de4-b7ca-f77ea4c8e6ea

### CODE REVIEW

File: app/lib/features/chat/widgets/custom_input.dart [hunk 1]
```diff
@@ -1040,6 +898,126 @@ class _TextInputWidget extends ConsumerWidget {
       chatInputNotifier.emojiPickerVisible(false);
     }
   }
+
+  @override
+  Widget build(BuildContext context) {
+    ref.listen(chatInputProvider(widget.roomId), (prev, next) {
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/custom_input.dart:904]: this was mostly a move, correct?

File: app/lib/features/chat/widgets/custom_input.dart [hunk 2]
```diff
@@ -1040,6 +898,126 @@ class _TextInputWidget extends ConsumerWidget {
...
+        chatFocus.requestFocus();
+      }
+    });
+    return CallbackShortcuts(
+      bindings: <ShortcutActivator, VoidCallback>{
+        const SingleActivator(LogicalKeyboardKey.enter): () {
+          widget.onSendButtonPressed();
+        },
+      },
+      child: MultiTriggerAutocomplete(
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/custom_input.dart:924]: only this part actually update, am I right?

File: app/lib/features/chat/widgets/bubble_builder.dart [hunk 3]
```diff
@@ -54,19 +57,19 @@ class BubbleBuilder extends ConsumerWidget {
         isMemberEvent
             ? child
             : SwipeTo(
-                onRightSwipe: (DragUpdateDetails details) {
-                  ref
-                      .read(chatInputProvider(roomId).notifier)
-                      .setReplyToMessage(message);
-                },
+                onRightSwipe: redactedOrEncrypted
```
Comment (gnunicorn) [inline app/lib/features/chat/widgets/bubble_builder.dart:60]: you should probably also mention this new feature in the description...

**Review by gnunicorn — APPROVED** [review]
Looking good. But some improvements in the usage of the consumers ... though it stands to reason, this can be applied in a follow-up-PR soon, too.

File: app/lib/features/chat/widgets/custom_input.dart (hunk 1 above)
Comment (gtalha07) [reply app/lib/features/chat/widgets/custom_input.dart]: Yep.

File: app/lib/features/chat/widgets/custom_input.dart (hunk 2 above)
Comment (gtalha07) [reply app/lib/features/chat/widgets/custom_input.dart:918]: Yes indeed.

**Pull request merged**